extension/chrome/settings/setup/setup-key-manager-autogen.ts

```diff
@@ -54,6 +54,7 @@ export class SetupWithEmailKeyManagerModule {
       if (ApiErr.isNetErr(e) && await Api.isInternetAccessible()) { // frendly message when key manager is down, helpful during initial infrastructure setup
         e.message = `FlowCrypt Email Key Manager at ${this.view.orgRules.getKeyManagerUrlForPrivateKeys()} is down, please inform your network admin.`;
       }
+      await Ui.modal.error((e as Error).message);
       throw e;
     }
   };
```

Comment:
We should either handle an error (with a modal) or throw it. It's weird to be doing both. Plus this change wasn't tested. I'll revert this PR.

Reply:
there is a test `fail@key-manager-server-offline.flowcrypt.test - shows friendly KM not reachable error` that covers this new modal

Reply:
That test was already there before. Since functionality was broken, I don't think it covered it.

Reply:
The lines below, already present before this PR, already expected a modal to show up:

![image](https://user-images.githubusercontent.com/6306961/157908802-67778dc6-8cc2-430e-82b2-90a4b32dc2bc.png)

And yet, the error was still present.

The existing test only covers situation with autogen (ekm not available during automatic setup) but the bug we encountered is during user-involved setup. Most likely, that uses different code.

Reply:
Oh I see. However, the original bug is not referenced here

Reply:
Let's have a look at #4334 once I'm ready

Reply:
It's a bit more complicated, even - the added line above checks for a modal, while the lines below are testing for a "view-rendered error" that is rendered into the page instead.

To my knowledge, the this PR:
 - in case of autogen, it would show both a modal and then also a whole-page error (that's why the test above is now testing for both, which is anyway wrong)
 - in case of import + require user to enter pass phrase (like flowcrypt.com has it), it would have showed a modal. But this case wasn't tested at all.

Cleaned all of this up in the alternative PR.